### PR TITLE
Maciej/integrate updater

### PR DESCRIPTION
**Pull request by maciej-flexcompute**
TODO:
1. convert to unit testing: [examples/testing_unit_system.py](https://github.com/flexcompute/Flow360/pull/104#diff-7c0f2a5e525ab67b3824045d569062513ef76c70589d779433ece312b41e3763)

2. Serialise UnitSystem to simpler expressions: SI, CGS, imperial, flow360.
3. Introduced raised errors does not use flow360 error handling with log.error - thus message printing is inconsistent.

### CODE REVIEW

File: examples/testing_unit_system.py
Comment (maciej-flexcompute) [inline examples/testing_unit_system.py:1]: remove this file if this is all implemented in tests

File: flow360/component/flow360_params/flow360_params.py [hunk 1]
```diff
@@ -22,9 +22,18 @@
...
+from ...exceptions import (
+    Flow360ConfigError,
+    Flow360NotImplementedError,
+    Flow360RuntimeError,
+    Flow360ValidationError,
+)
 from ...log import log
 from ...user_config import UserConfig
+
+# from .updater import update
```
Comment (maciej-flexcompute) [inline flow360/component/flow360_params/flow360_params.py:35]: remove this comment

File: flow360/component/flow360_params/flow360_params.py [hunk 2]
```diff
@@ -1470,74 +1451,115 @@ class Flow360Params(Flow360BaseModel):
...
+                    elif name == "CGS":
+                        kwarg_unit_system = CGSUnitSystem()
+                    elif name == "Imperial":
+                        kwarg_unit_system = ImperialUnitSystem()
+                    elif name == "Flow360":
+                        kwarg_unit_system = Flow360UnitSystem()
+                    else:
+                        raise Flow360RuntimeError(f"Undefined unit system name provided: {name}")
+                else:
+                    kwarg_unit_system = UnitSystem(**kwarg_unit_system)
```
Comment (maciej-flexcompute) [inline flow360/component/flow360_params/flow360_params.py:1475]: lets wrap this with a function in unit_system.py
```python
@classmethod
def UnitSystem.from_dict(cls, unit_system_dict):
    pass
```

File: flow360/component/flow360_params/flow360_params.py [hunk 3]
```diff
@@ -1470,74 +1451,115 @@ class Flow360Params(Flow360BaseModel):
...
+    @classmethod
+    def from_file(cls, filename: str) -> Flow360BaseModel:
+        """Loads a :class:`Flow360BaseModel` from .json, or .yaml file.
+
+        Parameters
+        ----------
+        filename : str
+            Full path to the .yaml or .json file to load the :class:`Flow360BaseModel` from.
+        **parse_obj_kwargs
+            Keyword arguments passed to either pydantic's ``parse_obj`` function when loading model.
```
Comment (maciej-flexcompute) [inline flow360/component/flow360_params/flow360_params.py:1511]: remove

File: flow360/component/flow360_params/flow360_params.py [hunk 4]
```diff
@@ -1470,74 +1451,115 @@ class Flow360Params(Flow360BaseModel):
...
+
+        Returns
+        -------
+        :class:`Flow360BaseModel`
```
Comment (maciej-flexcompute) [inline flow360/component/flow360_params/flow360_params.py:1515]: `Flow360Params`

File: flow360/component/flow360_params/flow360_params.py (hunk 2 above)
Comment (maciej-flexcompute) [reply flow360/component/flow360_params/flow360_params.py:1475]: why `_infer_context` cant be used here?

File: flow360/examples/om6wing/flow360.json [hunk 5]
```diff
@@ -1,90 +1,104 @@
...
+            "value": 1.0,
+            "units": "kg/m**3"
+        },
+        "viscosity": {
+            "value": 1.0,
+            "units": "Pa*s"
+        },
+        "angular_velocity": {
+            "value": 1.0,
+            "units": "rad/s"
```
Comment (maciej-flexcompute) [inline flow360/examples/om6wing/flow360.json:45]: I think we can serialise it to SI

File: tests/test_flow360_params.py [hunk 6]
```diff
@@ -286,12 +297,15 @@ def test_params_with_units():
...
-    with fl.UnitSystem(base_system=u.BaseSystemType.CGS, length=2.0 * u.cm):
-        params_reimport = fl.Flow360Params(**json.loads(params_as_json))
-        assert params_reimport.geometry.ref_area == params.geometry.ref_area
+    to_file_from_file_test(params)
+
+    # This is not supported anymore I presume?
+
+    # with fl.UnitSystem(base_system=u.BaseSystemType.CGS, length=2.0 * u.cm):
+    #    params_reimport = fl.Flow360Params(**json.loads(params_as_json))
+    #    assert params_reimport.geometry.ref_area == params.geometry.ref_area
```
Comment (maciej-flexcompute) [inline tests/test_flow360_params.py:308]: this is right, we can remove

File: tests/test_handle_version_and_unit_system.py [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+def test_parse_no_version():
+    with flow360.SI_unit_system:
+        params = Flow360Params(**params_no_version)
+    assert params.version == flow360.__version__
+
+
+# def test_parse_wrong_version():
+#     print(flow360.__version__)
+#     params = Flow360Params(**params_old_version)
+#     assert params.version == flow360.__version__
```
Comment (maciej-flexcompute) [inline tests/test_handle_version_and_unit_system.py:266]: should it raise notimplemented error?

File: flow360/examples/om6wing/flow360.json (hunk 5 above)
Comment (awkrupka) [reply flow360/examples/om6wing/flow360.json:45]: Yes, hasn't been updated yet since i think there was no comparison tests for it.

**Pull request merged**